### ISSUE TITLE
Additional lifted operators

### ISSUE THREAD

**Issue by Andrew-S-Rosen** [Parsl/parsl#2945]
**Is your feature request related to a problem? Please describe.**

Following up #2904, we should consider adding additional "lifted" operators based on the [special method names list](https://docs.python.org/3/reference/datamodel.html#special-method-names)

**Describe the solution you'd like**

More lifted operators!

**Describe alternatives you've considered**

No more lifted operators, but that would be sad 😢 

**Additional context**

We need to decide _which_ methods to implement and which to not implement. Actually implementing it and adding tests is trivial. I am happy to do that.

The main category to look at, in my opinion, is the list of [numeric types](https://docs.python.org/3/reference/datamodel.html#emulating-numeric-types). Which of these should we be implementing? There are a lot, and not all of them are necessarily worthwhile to include. 

My proposal at the moment is to include the following and then any others based on justified request:

```bash
object.__add__(self, other)
object.__sub__(self, other)
object.__mul__(self, other)
object.__matmul__(self, other)
object.__truediv__(self, other)
object.__floordiv__(self, other)
object.__mod__(self, other)
object.__divmod__(self, other)
object.__pow__(self, other[, modulo])
```

**Comment by Andrew-S-Rosen** [comment]
Side-note: I wonder if there is some way to support the following behavior using a similar approach:

```python
from parsl import python_app

@python_app
def test(a, b):
    c = a + 1
    d = b + 1
    return c, d

c_future, d_future = test(1, 2)
```

I wonder if users might expect this to be possible...

**Comment by benclifford** [comment]
I don't think this case is distinguishable from returning a tuple `(c,d)` that the user intends to be contained in a single future.

**Comment by Andrew-S-Rosen** [comment]
```python
from parsl import python_app

@python_app
def test(a, b):
    c = a + 1
    d = b + 1
    return (c, d)

future = test(1, 2)
```

The above always works, of course. But in "normal Python", one might expect to be able to do `c, d = test(1, 2)` instead, which won't work (`ValueError: too many values to unpack (expected 2)`). Not a big deal since it really is _a_ single future being returned, but I brought it up anyway.

Unless you meant something else?

**Comment by benclifford** [comment]
I think it would need the appfuture to be iterable (so as to generate Futures for each possible future element, to them be assigned to the receiving `c_future`, `d_future` variables) but I don't think the app future could be iterable because it doesn't know enough about the iterable-ness of the underlying value contained in the future: most specifically, I think, it doesn't know the number of elements to make entry-future for. (this is like for AppFutures-of-dicts, you can't ask for the list of available names using `keys()` - you can only ask for a name and wait until the future to discover if it exists or not)

**Comment by Andrew-S-Rosen** [comment]
Ah, that is a good point. You're right. The `length` isn't known _a priori_.

**Comment by mirembe-mariam** [comment]
hello @benclifford , can i work on this issue?

**Comment by Unique-Usman** [comment]
I think you can go ahead @mirembe-mariam  since I got occupied with some other stuffs.
